src/coreclr/src/tools/aot/ILCompiler.ReadyToRun/IL/Stubs/PInvokeILEmitter.cs

```diff
@@ -29,6 +29,12 @@ private PInvokeILEmitter(MethodDesc targetMethod)
         {
             Debug.Assert(targetMethod.IsPInvoke);
             _targetMethod = targetMethod;
+
+            // P/Invoke marked with UnmanagedCallersOnlyAttribute is not
```

Comment:
The other similar NotSupported checks are in EmitIL. Is there a reason why this check is here and not in EmitIL?

(just would like to know whether it is intentional or not)

Reply:
It was originally intentionally because I was going to handle it in a similar manner as the native side (i.e. during the reading of metadata). Looking at this now, I see what you are saying and since I didn't fold the check into `GetPInvokeMethodMetadata()` I think it should be there. Will move.